boot_services/src/boot_services.rs

```diff
@@ -310,14 +322,23 @@ pub trait BootServices: Sized {
         unsafe { self.uninstall_protocol_interface_unchecked(handle, protocol.protocol_guid(), interface_ptr) }
     }
 
+    /// # Safety
+    ///
+    /// Any user using `fn uninstall_protocol_interface_unchecked()` should ensure that the interface is of the right
+    /// type for the specified protocol before calling the function. Failure to do so could lead to unexpected runtime
+    /// behaviors.
     unsafe fn uninstall_protocol_interface_unchecked(
         &self,
         handle: efi::Handle,
         protocol: &'static efi::Guid,
         interface: *mut c_void,
     ) -> Result<(), efi::Status>;
 
-    fn reinstall_protocol_interface<P: Protocol<Interface = I> + 'static, I: 'static>(
+    /// # Safety
+    ///
+    /// Any user using `fn reinstall_protocol_interface()` should ensure that the efi::Handle pointer is a valid
+    /// pointer. Failure to do so could lead to unexpected runtime behaviors.
+    unsafe fn reinstall_protocol_interface<P: Protocol<Interface = I> + 'static, I: 'static>(

```

Comment:
should this not be `reinstall_protocol_interface_unchecked`?

To match the other functions, you'd probably also have a reinstall_protocol_interface

Reply:
Hi Joey,

Clippy flagged `reinstall_protocol_interface()` and `uninstall_protocol_interface()` with the following error:

```
error: this public function might dereference a raw pointer but is not marked `unsafe`
   --> boot_services\src/boot_services.rs:358:17
    |
358 |                 handle,
    |                 ^^^^^^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#not_unsafe_ptr_arg_deref
    = note: `#[deny(clippy::not_unsafe_ptr_arg_deref)]` on by default
```

To resolve this, I marked both functions as `unsafe`, even though we already have unchecked versions marked as `unsafe`.

If marking them as `unsafe` isn't the preferred solution, I can suppress the lint for these functions instead.

Reply:
I agree with clippy more than I agree with myself. Please follow clippy :) Thanks for double checking my request!